frontend/src/workspaces/features/DownloadBucketObject/DownloadBucketObject.tsx

```diff
@@ -32,9 +32,14 @@ const DownloadBucketObject = (props: DownloadBucketObjectProps) => {
 
   const onClick = async () => {
     setIsPreparing(true);
+    const openInNewTab = object.key.toLowerCase().endsWith(".html");
     try {
-      const url = await getBucketObjectDownloadUrl(workspace.slug, object.key);
-      await downloadURL(url);
+      const url = await getBucketObjectDownloadUrl(
+        workspace.slug,
+        object.key,
+        !openInNewTab,
+      );
+      await downloadURL(url, openInNewTab ? "_blank" : "");
```

Comment:
[Nitpick] I think the cherry on the cake would be changing the "Download" label on the button to "View" if `openInNewTab`. What do you think?